apps/prairielearn/elements/pl-file-editor/pl-file-editor.js

```diff
@@ -26,6 +26,94 @@ window.PLFileEditor = function (uuid, options) {
   this.editor.setReadOnly(options.readOnly);
   this.editor.getSession().on('change', this.syncFileToHiddenInput.bind(this));
 
+  this.session = this.editor.getSession();
+  this.doc = this.session.getDocument();
+  this.hasRanges = options.hasRanges;
+  this.rangeList = [];
+  
+
+  let editor = this.editor,
+      session = this.session,
+      doc = this.doc;
+
+function before(obj, method, wrapper) {
+  var orig = obj[method];
+  obj[method] = function() {
+    var args = Array.prototype.slice.call(arguments);
+    return wrapper.call(this, function(){
+      return orig.apply(obj, args);
+    }, args);
+  }
+
+  return obj[method];
+}
+
+function intersects(range) {
+  var cursor = editor.selection.getCursor();
+  return (
+    cursor.row >= range.start.row && cursor.row <= range.end.row &&
+    cursor.column >= range.start.column && cursor.column <= range.end.column
```

Comment:
I'm not convinced that allowing editing in parts of a line is needed.

Reply:
I have questions right now that I would rewrite to use the intra-line editing. It ends up being useful for questions where you ask a student to complete the line but do so in a constrained way. A simple example would be to force the student to use the ternary operator. As an example,

Complete the following function to return the lowercase version of `c` if it is an alphabetical character. If `c` is not alphabetical, just return `c`. You may use all functions from the `ctype.h` **EXCEPT FOR** `tolower` in your solution. Your solution **MUST** use the ternary operator. 

```c
char my_tolower(char c){
  return /*editable section 1*/ ? /*editable section 2*/;
}
```

And I want them to come up with

```c
char my_tolower(char c){
  return isupper(c) ? 'a' + (c - 'A') : c;
}
```

I know there are other ways currently to force/check for certain operations but this feels like a nice way to write the question for the instructor and a nice way to answer for the student.